typescript/union_types.go

```diff
@@ -0,0 +1,57 @@
+package typescript
+
+import (
+	"fmt"
+)
+
+// DeclareType instantiates a new type declaration
+func DeclareType(name string, typeRef Writable) *TypeDeclaration {
+	return &TypeDeclaration{
+		name:          name,
+		typeReference: typeRef,
+	}
+}
+
+// TypeDeclaration declares a type
+type TypeDeclaration struct {
+	name          string
+	typeReference Writable
+}
+
+// WriteCode writes the type declaration to the writer
+func (td *TypeDeclaration) WriteCode(writer CodeWriter) {
+	writer.Eol()
+	writer.Write(fmt.Sprintf("type %s = ", td.name))
+	td.typeReference.WriteCode(writer)
+	writer.Write(";")
+	writer.Eol()
+}
+
+// UnionTypeRef represents the union of a series of type refs
+// see: https://www.typescriptlang.org/docs/handbook/typescript-in-5-minutes.html#unions
+type UnionTypeRef struct {
+	typeRefs []Writable
+}
+
+// UnionType is a constructor function for creating union types
+func UnionType(refs ...string) *UnionTypeRef {
```

Comment:
Might as well add a method to mutate the union/product a new one

```
func (a *UnionTypeRef) Union(b Writable) *UnionTypeRef {
  // create union of a + b
}```

Reply:
this (https://github.com/Zocdoc/gopoetry/pull/14/commits/582e0c932c7abf17609dfc8df9452dfc29fd2649)  basically what you had in mind?